instrumentation/log4j/log4j-2.13.2/library/src/test/groovy/OpenTelemetryAppenderTest.groovy

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.instrumentation.log4j.v2_13_2.InMemoryLogExporter
+import io.opentelemetry.instrumentation.log4j.v2_13_2.OpenTelemetryAppender
+import io.opentelemetry.instrumentation.log4j.v2_13_2.OpenTelemetryLog4j
+import io.opentelemetry.instrumentation.log4j.v2_13_2.SimpleLogProcessor
+import io.opentelemetry.instrumentation.test.InstrumentationSpecification
+import io.opentelemetry.instrumentation.test.LibraryTestTrait
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo
+import io.opentelemetry.sdk.logging.LogSinkSdkProvider
+import io.opentelemetry.sdk.resources.Resource
+import org.apache.logging.log4j.LogManager
+
+class OpenTelemetryAppenderTest extends InstrumentationSpecification implements LibraryTestTrait {
+
+  private static InMemoryLogExporter logExporter
+  private static Resource resource
+  private static InstrumentationLibraryInfo instrumentationLibraryInfo
+  private static Attributes attributes
+
+  def setupSpec() {
+    logExporter = InMemoryLogExporter.create()
+    resource = Resource.getDefault()
+    instrumentationLibraryInfo = InstrumentationLibraryInfo.create(OpenTelemetryAppender.getName(), null)
+    attributes = Attributes.builder().put("logger.name", "TestLogger").put("thread.name", "Time-limited test").build()
+
+    def sdkSinkProvider = LogSinkSdkProvider.builder().build()
+    sdkSinkProvider.addLogProcessor(SimpleLogProcessor.create(logExporter))
+    def logSink = sdkSinkProvider.get(null, null)
+    OpenTelemetryLog4j.initialize(logSink, resource)
```

Comment:
I think this is a reasonable approach for now. Is it possible to build more complex configuration that would allow log4j to run the initialization itself?